### ISSUE TITLE
Killer on the Loose: keeping players close together

### ISSUE THREAD

**Issue by FTWinston** [KillerMinecraft/KillerMinecraft#171]
To prevent one survivor going it alone, we ideally want an in-game incentive for the players to stick together.

The most likely approach for this is to apply a slowly increasing penalty to players that have been far from all other players for some time.

Perhaps a minor penalty after 5 mins, increasing again after 10, and then increasing every minute thereafter. (These numbers are very provisional.)

What this penalty should be isn't entirely clear. Options include some/all of the following:
* Rapidly increasing hunger drain
* Slight damage, which gets more frequent and hits you harder
* a detrimental potion effect

Going it alone may still be possible (especially at later stages), but shouldn't seem easier than sticking together.

**Comment by FTWinston** [comment]
Perhaps we could tie this into the day-night cycle, since that is already a key element.

If an entire night is spent away from other players (and not in the nether), a big penalty is taken. Perhaps even you HAVE to be in a bed, with other players nearby?

**Comment by FTWinston** [comment]
I have added the following process, for Mystery Killer only:

Every 2.5 seconds, the distance from each player to each other player is calculated. (This is handled in a sensible two-way process, to avoid redundant checking.)

If a player is within 25 metres of another player, their "timer" is reset.

After a timer has run for 5 minutes without being reset, it applies the first status effect.

Every minute thereafter, it also applies the next status effect.

Status effects last 15 seconds and are renewed every 15 seconds, so players will likely end up loitering briefly near each other to be sure that effects have been removed.
